airbyte-cdk/java/airbyte-cdk/db-sources/src/main/kotlin/io/airbyte/cdk/components/debezium/StateFilesAccessor.kt

```diff
@@ -0,0 +1,192 @@
+    }
+
+    fun writeOffset(offset: DebeziumComponent.State.Offset) {
+        val bytesMap =
+            offset.debeziumOffset
+                .map { (key, value) -> Pair(toByteBuffer(key), toByteBuffer(value)) }
+                .toMap()
+        fileOffsetBackingStore.start()
+        fileOffsetBackingStore.set(bytesMap, null)
+        fileOffsetBackingStore.stop()
+    }
+
+    fun readSchema(): DebeziumComponent.State.Schema {
+        val builder = ImmutableList.builder<HistoryRecord>()
+        val consumer = Consumer { element: HistoryRecord -> builder.add(element) }
+        fileSchemaHistory.start()
+        try {
+            RECOVER_RECORDS_METHOD.invoke(fileSchemaHistory, consumer)
+        } catch (e: IllegalAccessException) {
+            throw RuntimeException(e)
+        } catch (e: InvocationTargetException) {
+            throw RuntimeException(e)
+        }
+        fileSchemaHistory.stop()
+        return DebeziumComponent.State.Schema(builder.build())
+    }
+
+    fun writeSchema(schema: DebeziumComponent.State.Schema) {
+        fileSchemaHistory.initializeStorage()
+        fileSchemaHistory.start()
+        for (r in schema.debeziumSchemaHistory) {
+            try {
+                STORE_RECORD_METHOD.invoke(fileSchemaHistory, r)
+            } catch (e: IllegalAccessException) {
+                throw RuntimeException(e)
+            } catch (e: InvocationTargetException) {
+                throw RuntimeException(e)
+            }
+        }
+        fileSchemaHistory.stop()
+    }
+
+    init {
+        try {
+            workingDir = Files.createTempDirectory(Path.of("/tmp"), "airbyte-debezium-state")
+        } catch (e: IOException) {
+            throw UncheckedIOException(e)
+        }
+        offsetFilePath = workingDir.resolve("offset.dat")
+        schemaFilePath = workingDir.resolve("dbhistory.dat")
+        // Create and configure FileOffsetBackingStore instance.
+        fileOffsetBackingStore = FileOffsetBackingStore(keyConverter())
+        val offsetProps = HashMap<String, String>()
+        offsetProps[WorkerConfig.KEY_CONVERTER_CLASS_CONFIG] = JsonConverter::class.java.name
+        offsetProps[WorkerConfig.VALUE_CONVERTER_CLASS_CONFIG] = JsonConverter::class.java.name
+        offsetProps[StandaloneConfig.OFFSET_STORAGE_FILE_FILENAME_CONFIG] =
+            offsetFilePath.toString()
+        fileOffsetBackingStore.configure(StandaloneConfig(offsetProps))
+        // Create and configure FileSchemaHistory instance.
+        fileSchemaHistory = FileSchemaHistory()
+        val schemaConfig =
+            Configuration.create()
+                .with(FileSchemaHistory.FILE_PATH, schemaFilePath.toString())
+                .build()
+        fileSchemaHistory.configure(
+            schemaConfig,
+            HistoryRecordComparator.INSTANCE,
+            SchemaHistoryListener.NOOP,
+            false
+        )
+    }
+
+    companion object {
```

Comment:
why the need for a companion object? Looks like there's one object per instance of StateFileAccessor, and we access its fields indiscriminately. What am I missing? Might be a kotlin idiom?

Reply:
It's the only way to have static methods, but it seems that I forgot to decorate these with @JvmStatic. Will revisit!